components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java

```diff
@@ -508,11 +562,18 @@ public void addScope(String scope, String[] claims) throws IdentityOAuthAdminExc
      */
     public void deleteScope(String scope) throws IdentityOAuthAdminException {
 
+        validateScopeName(scope);
+        // Check whether a scope exists with the provided scope name which to be deleted.
+        validateScopeExistence(scope);
```

Comment:
We might not be able to successfully say that scope does not exist from this validation. In a concurrent scenario, if the resource is existing while executing this check and while trying to delete the resource actually, it might have been deleted from another delete request. While it does not break any functionality, it might introduce inconsistencies.

Reply:
yes, there are cases that we need to handle the concurrent scenarios properly. As per the offline discussion for the moment keep as it is and we'll address this later.